test-requirements.txt

```diff
@@ -2,6 +2,7 @@ pytest==4.4.1
 pytest-xdist==1.26.1
 pytest-cov==2.6.1
 nbval==0.9.3
+IPython>=3.2.1
```

Comment:
Added to `test-requirements.txt` rather than `requirements.txt` since `nbval` has an `IPython` dependency, causing the `core-dependencies=True` test to fail. @dsherry preferred it to be in `requirements.txt`, but this change made the tests pass. Input on this is welcome.

Reply:
I think this is ok!